storage/innobase/lock/lock0lock.cc

```diff
@@ -4583,12 +4588,49 @@ static bool lock_release_on_prepare_try(trx_t *trx)
       if (latch->try_acquire())
       {
         if (!rec_granted_exclusive_not_gap)
+        {
           lock_rec_dequeue_from_page(lock, false);
+          latch->release();
+        }
         else if (supremum_bit)
+        {
           lock_rec_unlock(*cell, lock, PAGE_HEAP_NO_SUPREMUM);
+          latch->release();
+        }
         else
-          lock_rec_unlock_unmodified(*cell, lock, offsets, heap, mtr);
-        latch->release();
+        {
+          ulint set_nth_bit_calls= trx->lock.set_nth_bit_calls;
+          latch->release();
+          lock_sys.rd_unlock();
+          trx->mutex_unlock();
+          mtr.start();
+          buf_block_t *block= btr_block_get(
+              *lock->index, lock->un_member.rec_lock.page_id.page_no(),
+              RW_S_LATCH, true, &mtr, nullptr, nullptr,
+              BUF_GET_POSSIBLY_FREED);
```

Comment:
Could we set a new debug-only field in `mtr` to signal that `btr_block_get()` should not complain if the page has been freed meanwhile?